src/extensions/replay/sessionrecording.ts

```diff
@@ -579,12 +590,11 @@ export class SessionRecording {
         this.windowId = windowId
         this.sessionId = sessionId
 
-        if (
-            returningFromIdle ||
-            ([FULL_SNAPSHOT_EVENT_TYPE, META_EVENT_TYPE].indexOf(event.type) === -1 &&
-                (windowIdChanged || sessionIdChanged || isUndefined(this._fullSnapshotTimer)))
-        ) {
-            this._tryTakeFullSnapshot()
+        if (sessionIdChanged || windowIdChanged) {
+            this.stopRecording()
+            this.startIfEnabledOrStop()
```

Comment:
Haven't looked into it but wondering if we need to consider anything here with regards to ingestion controls (sampling, feature flags, minimum duration) that might affect the new recording not starting

Reply:
sampling and feature flags should just work...

min duration is interesting... i don't know if we'd restart the counter - we should... 👀

Reply:
yep... session start is read from storage, most recent snapshot data from the buffer, so we're safe with min duration  too 👍